cfme/tests/cloud/test_providers.py

```diff
@@ -36,6 +46,107 @@ def enable_regions(provider):
     enable_provider_regions(provider)
 
 
+@pytest.fixture(scope='function')
+def az_pwsh_vm(provider, appliance, request):
+    # azure_pwsh contains powershell and necessary scripts
+    vm_name = random_vm_name(context="pwsh")
+    template = provider.data.get('powershell_vm')
+    vm = provider.appliance.collections.infra_vms.instantiate(vm_name, provider, template)
```

Comment:
Why infra_vms collection here, for what I'm assuming is a fixture used by tests parametrized on Azure?

You can use `appliance.provider_based_collection` to get the right collection type.

As a second note, although the ProviderFilter required_fields is used to only parametrize on providers with 'powershell_vm', I think it would be good to wrap that reference in a try/except that would skip the test should this fixture be used in other test functions that forget to include this required_field.

Reply:
@mshriver right, I started to mess around with that because I wanted to provision a preallocated disk, however, it appears to be not supported in RHV, so I'm reverting to my original implementation